trl/trainer/online_dpo_trainer.py

```diff
@@ -1,590 +1,483 @@
 
-    def generate_completions(self, sampling: bool = False):
-        args = self.args
-        tokenizer = self.tokenizer
-        generation_config = GenerationConfig(
-            max_new_tokens=self.args.response_length,
-            temperature=(0.01 + 1e-7),
-            top_k=0.0,
-            top_p=1.0,
-            do_sample=True,
-        )
-
-        table = defaultdict(list)
-        with unwrap_model_for_generation(self.model, self.accelerator) as unwrapped_model:
-            for batch in self.eval_dataloader:
-                query = batch["input_ids"]
-                with torch.no_grad():
-                    context_length = query.shape[1]
-                    query_response, _ = batch_generation(
-                        unwrapped_model,
-                        query,
-                        query.shape[0],
-                        tokenizer.pad_token_id,
-                        generation_config,
-                    )
-                    response = query_response[:, context_length:]
-                    postprocessed_response = response
-                    if args.stop_token_id is not None:  # handle the edge case when stop_token_id exists but is 0
-                        postprocessed_response = truncate_response(
-                            args.stop_token_id, tokenizer.pad_token_id, response
-                        )
-                    # For the WandB table of prompts and completions, we keep
-                    # the special tokens associated with the chat template, but
-                    # discard the padding tokens for readability
-                    table_query = [
-                        q.replace(tokenizer.pad_token, "")
-                        for q in tokenizer.batch_decode(query, skip_special_tokens=False)
-                    ]
-                    table_response = [
-                        r.replace(tokenizer.pad_token, "")
-                        for r in tokenizer.batch_decode(postprocessed_response, skip_special_tokens=False)
-                    ]
-                    table["query"].extend(gather_object(table_query))
-                    table["model response"].extend(gather_object(table_response))
-
-                    postprocessed_query_response = torch.cat((query, postprocessed_response), 1)
-                    _, score, _ = get_reward(
-                        self.reward_model, postprocessed_query_response, tokenizer.pad_token_id, context_length
-                    )
-                    table["score"].extend(self.accelerator.gather(score).float().cpu().numpy())
-
-                if sampling:
-                    break
-        df = pd.DataFrame(table)
-
-        if self.accelerator.is_main_process:
-            print_rich_table(df.iloc[0 : 0 + 5])
-            if "wandb" in args.report_to:
-                import wandb
-
-                if wandb.run is not None:
-                    wandb.log({"completions": wandb.Table(dataframe=df)})
+    def empty_cache(self):
```

Comment:
This could be moved to utils?

Reply:
also note this helper: https://github.com/huggingface/trl/commit/1e4fb80cbc8b0ad688269f90eac8282203410620

Reply:
Done in https://github.com/huggingface/trl/pull/1839/commits/57eb67338e4a20d41ab6fb33a801fd973bca10c5